c-api/compat-5.3.c

```diff
@@ -336,23 +348,132 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (!defined (__MINGW32__) && defined(__GNUC__) && (__GNUC__ < 6))
+    /* use strerror_r here, because it's available on these specific platforms */
+    char buf[512] = {0};
+#if ((defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) || (defined(_XOPEN_SOURCE) || _XOPEN_SOURCE >= 600)) && (!defined(_GNU_SOURCE) || !_GNU_SOURCE)
+    /* XSI Compliant */
+    strerror_r(en, buf, 512);
+    s = buf;
+#else
+	/* GNU-specific which returns const char* */
+    s = strerror_r(en, buf, 512);
+#endif
+#elif defined(_MSC_VER) || (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L)
```

Comment:
`strerror_s` is not a compulsory part of the standard (and infact, glibc doesn't provide it). Looking at  https://stackoverflow.com/a/17960590/282536 is sounds like an additional check of `__STDC_LIB_EXT1__` would be required.